frontend/src/components/Table/Table.tsx

```diff
@@ -96,6 +96,8 @@ export interface TableProps<T extends object = {}> extends TableOptions<T> {
   detailsPanel?: DetailsOptions<T>;
   footer?: boolean;
   hideToolbar?: boolean;
+  // Restrict where you would like an expansion button to appear
+  restrictExpansion?: (val: any) => boolean;
```

Comment:
I think I'd prefer if this was named in the positive. Something like `canRowExpand?: (row: any) => boolean`
What do you think? 
It's usually harder the other way: "is it not restrictExpansion that allows expansion?" _ugh_

Reply:
Also I think your logic below might not reflect the "restrict" naming...

Reply:
I agree, was having trouble naming. Started with showFolderWhere -> allowExpansionWhere -> restrictExpansion lol. Changing now